### PR TITLE
Create the Pull request template

### LINKED ISSUE #578: Create PR and Issue templates for nailgun
* PR should follow a template:
a) Summary - Description of change
b) Issue Type - Bug Fix or new addition/enhancement
c) Component - e.g. Activation key/Repository etc.

### PR DESCRIPTION
**Pull request by san7ket**
Fixes #578

**Point to the API documentation/Plugin/feature? Please describe.**

A clear and concise description of the plugin you have added or the entity you have updated

**Presentation that this code works**

An Example that the code works. You can exercise the code as raw API calls or using any other method

`Eg: # Read Snapshot
       Snapshot(host=<host_id>, id=<snapshot_id>).read()`

~~~~~~~~
**Additional Information**

### CODE REVIEW

**Comment by ntkathole** [comment]
[test]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+**Point to the API documentation/Plugin/feature? Please describe.**
+
+A clear and concise description of the plugin you have added or the entity you have updated
+
+**Presentation that this code works**
+
+An Example that the code works. You can exercise the code as raw API calls or using any other method
+
+`Eg: # Read Snapshot
```
Comment (mirzal) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think you want fenced code block, like this:

~~~
```
Eg: # Read Snapshot
    Snapshot(host=<host_id>, id=<snapshot_id>).read()
```
~~~

With only single `` ` ``, this code will be all in one line

**Comment by san7ket** [comment]
@mirzal do you think we should had anything else to PR template that would help in faster reviews

**Comment by mirzal** [comment]
We can probably make it little more welcoming, and point out to our docs for overview:

-----------------------------------

Thanks for contributing to Nailgun!

To ensure quick review and merge of your PR, please make sure that you followed Contributing instructions in `docs/index.rst`. Especially, include following information:

* **Point to the API/Plugin/feature documentation**

A clear and concise description of the plugin you have added or the entity you have updated

* **Demonstration that this code works**

An Example that the code works. You can exercise the code as raw API calls or using any other method

```
Eg:  # Read Snapshot
    Snapshot(host=<host_id>, id=<snapshot_id>).read()
```

**Comment by san7ket** [comment]
@sean797 @evgeni thoughts?

**Comment by evgeni** [comment]
Two quick thoughts:
1. You use formatted text instead of headings. The output is similar, but using headings might be better for parsing programmatically (think screen readers and a possible PR bot)
2. I'd like to have sections for the PR type (feature/fix/docs) and "affected entity" both help quickly scanning what the PR is about and again allows a possible PR bot to act on the (meta)data.

See https://github.com/ansible/ansible/blob/devel/.github/PULL_REQUEST_TEMPLATE.md for an example

**Comment by ntkathole** [comment]
Can one of the admins verify this patch?

**Comment by ntkathole** [comment]
ok to test

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+**Point to the API documentation/Plugin/feature? Please describe.**
```
Comment (swadeley) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Please don't use slash in text. Its ambiguous at best. Style guides say not to use "please".
s|Point to the API documentation/Plugin/feature? Please describe.|Point to the API documentation, Plugin, or feature and describe it.|

**Comment by san7ket** [comment]
@SatelliteQE/quality-engineers Can someone take this up

**Comment by mshriver** [comment]
@san7ket @SatelliteQE/quality-engineers I'll take this over and push commits to update.

**Review by JacobCallahan — DISMISSED** [review]
ACK

**Review by ogajduse — APPROVED** [review]
LGTM. I see no reason to block this PR.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+##### Description of changes
+
+Describe in detail the changes made, and any potential impacts to callers.
+
+##### Upstream API documentation, plugin, or feature links
+
+Link to any relevant upstream API documentation that relates to the content of the PR
```
Comment (swadeley) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
Link to any relevant upstream API documentation that relates to the content of the PR.
```

**Review by swadeley — APPROVED** [review]
ACK

**Review by JacobCallahan — APPROVED** [review]

**Review by swadeley — APPROVED** [review]
ACK

**Pull request merged**